odk1-src/form-functions.rst

```diff
@@ -0,0 +1,719 @@
+  \.\., current question's parent group, position(..), Used with :func:`position` to get the iteration index.
+  
+.. _boolean-operators:
+  
+Boolean operators
+------------------
+
+.. csv-table::
+  :header: , Explanation, Example
+
+  and, ``True`` if the expressions before and after are ``True``, ${age} > -1 and ${age} < 120
+  or, ``True`` if either of the expressions before or after are ``True``, ${age} < 19 or ${age} > 64
+  
+.. _comparison-operators:
+  
+Comparison operators
+-----------------------
+  
+.. csv-table::
+  :header: , Explanation, Example, Notes
+  
+  =, equal to, ${enrolled} = 'yes', Can compare numbers or strings.
+  !=, not equal to, ${enrolled} != 'yes', Can compare numbers or strings.
+  >, greater than, ${age} > 17, 
+  >=, greater than or equal to, ${age} >= 18,
+  <, less than, ${age} < 65, 
+  <=, less than or equal to, ${age} <= 64,
+
+.. _math-operators:
+  
+Math operators
+---------------
+
+.. csv-table::
+  :header: , Explanation, Example, Notes
+  
+  \+, addition, ${salary_income} + ${self_employed_income}, Numbers only; does not concatenate strings.
+  \-, subtraction, ${income} - ${expenses},
+  \*, multiplication, ${bill} * 1.18,
+  div, division, ${percent_int} div 100, 
+  mod, `modulo`_ (division remainder), (${even_number} mod 2) = 0, 
+
+.. _modulo: https://en.wikipedia.org/wiki/Modulo_operation
+
+  
+  
+.. _functions:
+  
+Functions
+===========
+    
+.. _control-flow-functions:
+
+Control flow
+--------------
+
+.. function:: if(expression, then, else)
+
+  Returns :arg:`then` if :arg:`expression` evaluates to ``True``. 
+  Otherwise, returns :arg:`else`.
+
+  
+  
+.. function:: position(xpath)
+
+  Returns an integer equal to the 1-indexed position of the current node
+  within the node defined by :arg:`xpath`.
+  
+  Most often this is used in the form :tc:`position(..)`
+  to identify the current iteration index
+  within a repeat group.  
+  
+  .. container:: details
+  
+    .. include:: incl/form-examples/parallel-repeat-groups.rst
+
+.. function:: once(expression)
+
+  Returns the value :arg:`expression` if the question's value is empty.
+  Otherwise, returns the current value of the question.
+
+  This can be used to ensure that a random number is only generated once,
+  or to store the first value entered for a question
+  in a way that is retrievable even of the response is changed later.  
+
+.. _response-access-functions:
+  
+Accessing response values
+--------------------------
+
+.. note::
+
+  The response from most question types
+  can be accessed using :ref:`variables <variables>`.
+  Functions are needed for accessing responses to 
+  :ref:`multi select questions <select-functions>` and
+  questions inside :ref:`repeat groups <repeat-functions>`.
+
+.. _select-functions:
+  
+Select questions
+~~~~~~~~~~~~~~~~~~~
+
+.. function:: selected(select_question, choice_name)
+
+  Returns ``True`` if :arg:`choice_name` 
```

Comment:
I think it's useful to describe this a little more generally and mention that it searches for the second parameter in the space-separated list specified by the first parameter. There are all kinds of creative ways to use `selected` and `selected-at` with that more general concept.